packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1950,7 +1923,7 @@ export class TransactionController extends BaseController<
     );
 
     this.update((state) => {
-      state.transactions = this.#trimTransactionsForState(newTransactions);
```

Comment:
We still need to trim the transactions according to the transaction state limit?